### PR TITLE
Reduce default max network connection limit from 256 to 48 in bun install

### PR DESCRIPTION
**Pull request by Jarred-Sumner**
### What does this PR do?

This reduces the default max HTTP/HTTPS connection limit from 256 to 48 in bun install.

256 simultaneous HTTP requests is more than enough to saturate networks

When using a proxy, the default limit is 16 instead of 48.

Fixes https://github.com/oven-sh/bun/issues/4066

We already automatically adjust when there's a network error

### How did you verify your code works?

There are tests

### CODE REVIEW

**Comment by robobun** [comment]
@Jarred-Sumner, your commit a0f283b8b7cb8be83d0508885c4a99c1350ca627 is building: [`#5189`](https://buildkite.com/bun/bun/builds/5189)

File: test/cli/install/bun-install.test.ts [hunk 1]
```diff
@@ -67,20 +67,149 @@ afterAll(dummyAfterAll);
...
+});
+
 it("should not error when package.json has comments and trailing commas", async () => {
   const urls: string[] = [];
   setHandler(dummyRegistry(urls));
   await writeFile(
     join(package_dir, "package.json"),
     `
     {
-      // such comment!
```
Comment (alex-demchenko) [inline test/cli/install/bun-install.test.ts:77]: deleting the comment makes this test meaningless

Comment (Jarred-Sumner) [reply test/cli/install/bun-install.test.ts:77]: Thanks, didn’t mean to commit deleting this comment

File: src/install/install.zig [hunk 2]
```diff
@@ -9290,6 +9326,7 @@ pub const PackageManager = struct {
         clap.parseParam("--backend <STR>                       Platform-specific optimizations for installing dependencies. " ++ platform_specific_backend_label) catch unreachable,
         clap.parseParam("--registry <STR>                      Use a specific registry by default, overriding .npmrc, bunfig.toml and environment variables") catch unreachable,
         clap.parseParam("--concurrent-scripts <NUM>            Maximum number of concurrent jobs for lifecycle scripts (default 5)") catch unreachable,
+        clap.parseParam("--network-concurrency <NUM>           Maximum number of concurrent network requests (default 48)") catch unreachable,
```
Comment (Electroid) [inline src/install/install.zig:9329]: `--concurrent-requests` would go nicely with the existing flags

Comment (Jarred-Sumner) [reply src/install/install.zig:9329]: `--network-concurrency` is what pnpm does

File: src/install/install.zig [hunk 3]
```diff
@@ -1,3 +1,11 @@
+// Default to a maximum of 64 simultaneous HTTP requests for bun install if no proxy is specified
+// if a proxy IS specified, default to 16
+// https://github.com/npm/cli/issues/7072
+// https://pnpm.io/npmrc#network-concurrency (pnpm defaults to 16)
+// https://yarnpkg.com/configuration/yarnrc#networkConcurrency (defaults to 50)
+const default_max_simultaneous_requests_for_bun_install = 48;
+const default_max_simultaneous_requests_for_bun_install_for_proxies = 64;
```
Comment (KieranP) [inline src/install/install.zig:7]: Comment says 16, code sets to 64, which higher than non-proxy.

File: src/install/install.zig [hunk 4]
```diff
@@ -1,3 +1,11 @@
+// Default to a maximum of 64 simultaneous HTTP requests for bun install if no proxy is specified
```
Comment (KieranP) [inline src/install/install.zig:1]: Comment says 64, code sets 48.

**Pull request merged**